### PR TITLE
Bug/fixes 21

### PR DESCRIPTION
**Pull request by chrisbrantley**
To maintain consistency with filter params this fix converts '__' to '.' in swagger spec for sort params and also parses sort params to support the dot-separated format.

Fixes this issue: https://github.com/Vacasa/drf-jsonapi/issues/21

### CODE REVIEW

File: drf_jsonapi/serializers/resources.py [hunk 1]
```diff
@@ -432,10 +432,11 @@ def sort(cls, sort_param=None, queryset=None):
         if not sort_param:
             return queryset
 
-        sort_fields = list(filter(None, sort_param.split(',')))
+        sort_fields = list(filter(None, sort_param.replace('.', '__').split(',')))
 
         # validate the sort fields actually exist in the model
-        field_names = [field.name for field in queryset.model._meta.get_fields()]
+        field_names = getattr(cls.Meta, 'sort_fields', cls.Meta.fields)
+        field_names = list(field_names) + ['id']
```
Comment (ddobson) [inline drf_jsonapi/serializers/resources.py]: This is the same code you touched here: https://github.com/Vacasa/drf-jsonapi/pull/23/files#diff-994122ce09be62e5a94c8569a02a465bR439

Which of these is correct?

Comment (ddobson) [reply drf_jsonapi/serializers/resources.py]: @chrisbrantley ^

Comment (chrisbrantley) [reply drf_jsonapi/serializers/resources.py]: The two bits of code essentially do the same thing just in different ways. I thought the version in  #23 was slightly more pythonic.

**Comment by chrisbrantley** [comment]
@razzlepdx @morgandenning @ddobson This PR should be ready to approve and merge in. It was previously approved but was missing a lot of test coverage. I refactoring the tests to get us from 36% to 97% so hopefully that is sufficient? :)

**Pull request merged**